### DISCUSSION TITLE
Switching between light and dark themes

### DISCUSSION THREAD

**Discussion by alephnull** [kovidgoyal/kitty#7343] [Ideas]
Its more than just the kitty theme. Niggles I've had to hack around:
- LS_COLORS
- prompt theme
- completion zstyle

If the current theme name is exposed an an env variable, it might be enough.

**Comment by kovidgoyal** [comment]
The current theme cannot be exposed as an env var. env vars cannot be
updated in already launched processes. If you want to query whether
kitty is using a light or dark theme, use kitten @ get-colors

**Comment by alephnull** [comment]
Thanks! Doing it with get-colors would make the experience fully dynamic and I will not have to discard the current tabs when I make the switch as I do now. I would have to run kitten on every prompt render.

The above will work for the prompt theme which need the just the foreground and background colors to change. However for LS_COLORS, I will have to generate a theme based on the colors. [vivid](https://github.com/sharkdp/vivid) can generate LS_COLORS but it requires a theme name. Since the themes I switch between are known, I could do this ahead of time.

As I wrote this out, I realise that it is probably better to handle this outside kitty and change the kitty theme via remote control. For completeness, Chromium has `chrome://flags/#enable-force-dark` to make everything dark but enabling it requires a restart.

Kudos to you on taking terminals in an exciting new direction and thank you for being the 🦆

**Comment by Osx22** [comment]
I'm using kitty on arch.  Using kitten themes I've selected both a preferred "dark" and "light" theme.   Kitty detects the OS mode dark/light correctly on startup.   When I'm in Light switching to dark (gnome) kitty immediately adjusts and selects my preferred dark theme.   The other way around, when I am in dark and switch back to light?  Kitty doesn't adapt.   I created a small script as work around.  I've noticed that `gsettings get org.gnome.desktop.interface color-scheme`.  Give me 2 values.  'default' and 'prefer-dark'.

**Comment by kovidgoyal** [comment]
On Sat, Aug 23, 2025 at 06:39:19PM -0700, Osx22 wrote:
> I'm using kitty on arch.  Using kitten themes I've selected both a preferred "dark" and "light" theme.   Kitty detects the OS mode dark/light correctly on startup.   When I'm in Light switching to dark (gnome) kitty immediately adjusts and selects my preferred dark theme.   The other way around, when I am in dark and switch back to light?  Kitty doesn't adapt.   I created a small script as work around.  I've noticed that `gsettings get org.gnome.desktop.interface color-scheme`.  Give me 2 values.  'default' and 'prefer-dark'.

GNOME, as always is a special snowflake, read the note at
https://sw.kovidgoyal.net/kitty/kittens/themes/#change-color-themes-automatically-when-the-os-switches-between-light-and-dark
